### PR TITLE
telemetry: Metrics for amazonq unit test generation

### PR DESCRIPTION
**Pull request by laileni-aws**
## Problem
- No metrics for amazonq unit test generation
## Solution
Planning to add 4 metrics for Unit test generation.
1. amazonq_unitTestGeneration(new)
2. amazonq_utg_generateTests(new)
3. amazonq_utg_buildLoop(new)
4. ui_click(existing)

- Planning to add a new optional field (step) to ui_click to get metrics on which stage user has clicked a button.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ashishrp-aws — APPROVED** [review]

File: telemetry/definitions/commonDefinitions.json [hunk 1]
```diff
@@ -1457,6 +1497,31 @@
...
+        {
+            "name": "numberOfBuildFailed",
+            "type": "int",
+            "description": "Number of build failures in unit test generation build loop"
+        },
+        {
+            "name": "numberOfBuildPassed",
+            "type": "int",
+            "description": "Number of build succeeded in unit test generation build loop"
+        },
```
Comment (justinmk3) [inline telemetry/definitions/commonDefinitions.json:1514]: there are already some generic "successCount" and similar fields. use those. if you need to add a `executedCount` or something like that, that is preferred.

do not add over-specific fields for every metric.

File: telemetry/definitions/commonDefinitions.json [hunk 2]
```diff
@@ -1457,6 +1497,31 @@
...
+        {
+            "name": "numberOfUnitTestCasesAccepted",
+            "type": "int",
+            "description": "Number of unit test cases accepted"
+        },
+        {
+            "name": "numberOfUnitTestCasesGenerated",
+            "type": "int",
```
Comment (justinmk3) [inline telemetry/definitions/commonDefinitions.json:1522]: use or create generic `fooCount` metrics.

please always do this in the future. I have mentiond this in previous reviews.

File: telemetry/definitions/commonDefinitions.json [hunk 3]
```diff
@@ -1670,11 +1735,21 @@
...
         {
             "name": "successCount",
             "type": "int",
             "description": "The number of successful operations"
         },
+        {
+            "name": "supportedLanguage",
+            "type": "boolean",
+            "description": "supported language"
+        },
```
Comment (justinmk3) [inline telemetry/definitions/commonDefinitions.json:1752]: no.

File: telemetry/definitions/commonDefinitions.json (hunk 2 above)
Comment (laileni-aws) [reply telemetry/definitions/commonDefinitions.json:1522]: Named these two fields to common`acceptedCount` and `generatedCount`.

File: telemetry/definitions/commonDefinitions.json (hunk 3 above)
Comment (laileni-aws) [reply telemetry/definitions/commonDefinitions.json:1752]: I didn't see a field which checks the supported language(boolean).
LMK If I am missing something.

File: telemetry/definitions/commonDefinitions.json [hunk 4]
```diff
@@ -1694,6 +1749,16 @@
             "type": "string",
             "description": "Generic name of a template"
         },
+        {
+            "name": "testGenerationJobGroupName",
+            "type": "string",
+            "description": "Unit test generation job group name"
+        },
+        {
+            "name": "testGenerationJobId",
```
Comment (justinmk3) [inline telemetry/definitions/commonDefinitions.json:1758]: a group name and job id could be used in many metrics. please do not add hyper-specific fields for every metric.

I already see fields like `codeTransformJobId`, and that's another case where we're paying the price for those fields being over-specific.

```suggestion
            "name": "jobGroup",
            "type": "string",
            "description": "Job group name associated with the metric"
        },
        {
            "name": "jobId",
```

File: telemetry/definitions/commonDefinitions.json (hunk 3 above)
Comment (chandankmishra) [reply telemetry/definitions/commonDefinitions.json:1732]: This field should be isSupportedLanguage to indicate it is boolean field. SupportedLanguage seems like we are going to pass the actual value of language which could be string/enum type.

Also the comment should be "Indicate if the language is supported"

File: telemetry/definitions/commonDefinitions.json [hunk 5]
```diff
@@ -1417,6 +1442,26 @@
...
+            "type": "int",
+            "description": "Time from start to end of Generating unit tests"
+        },
+        {
+            "name": "linesOfCodeAccepted",
+            "type": "int",
+            "description": "Number of lines of code accepted"
+        },
+        {
+            "name": "linesOfCodeGenerated",
```
[the remark this answers is not in the document]
Comment (laileni-aws) [reply telemetry/definitions/commonDefinitions.json]: They are using `cwsprChatAcceptedNumberOfLines` which is specific name not generic but trying to use the same.

File: telemetry/definitions/commonDefinitions.json (hunk 4 above)
Comment (laileni-aws) [reply telemetry/definitions/commonDefinitions.json]: Will update with this

File: telemetry/definitions/commonDefinitions.json (hunk 3 above)
Comment (laileni-aws) [reply telemetry/definitions/commonDefinitions.json]: I did not see convention naming isXXXX for boolean values in the json file so for keeping consistent went with `supportedLanguage`
Let me change this accordingly

**Pull request closed without merging**